File: front/src/models/PropositionAchat.ts
import { ETypeObjetProposer } from "@enums/ETypeObjetProposer"

export type PropositionAchat =
{
    id: number,
    auteur: string,
    liste: ObjetProposer[]
}

export type DecisionAchatRequete =
{
    idPropositionAchat: number,
    idType: number | null,
    type: ETypeObjetProposer | null,
    achatEstValider: boolean
}

export type ObjetProposerRequete =
{
    type: ETypeObjetProposer,
    idType: number,
    quantite: number,
    idVaisseau: number | null,
    idStockage: number | null
}

export type ObjetProposer =
{
    nom: string,
    prixUnitaire: number,
    quantite: number,
    type: ETypeObjetProposer,
    idType: number,
    nomVaisseau: string,
    nomStockage: string
}

File: back/back/Models/PropositionAchat.cs
using System.Text.Json.Serialization;
using back.Enums;
using LiteDB;

namespace back.Models;

public sealed class PropositionAchat
{
     [BsonId]
     public int Id { get; set; }

     [BsonRef]
     public Personnage Personnage { get; set; } = null!;
     public List<ObjetProposer> Liste { get; set; } = [];
}

public sealed class ObjetProposer
{
    public int IdType { get; set; }
    public int? IdVaisseau { get; set; }
    public int? IdStockage { get; set; }
    public ETypeObjetProposer Type { get; set; }
    public string Nom { get; set; } = null!;

    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
    public string? NomVaisseau { get; set; }
    public int PrixUnitaire { get; set; }
     public int Quantite { get; set; }
}
